src/components/study-container.js

```diff
@@ -304,6 +305,92 @@ export function StudyContainer({ view, onChangeTab }) {
         }
     }, [studyUuid, loadTree]);
 
+    useEffect(() => {
+        const appName = 'GridStudy';
+        const MAX_TITLE_LENGTH = 106;
+
+        if (studyUuid) {
+            fetchElementAndParentsInfo(studyUuid)
+                .then((response) => {
```

Comment:
May be separate gathering of the response from building of the title from the response, in its own function.